src/ParamFile.cpp

```diff
@@ -0,0 +1,273 @@
+#include <algorithm>
+#include <cctype>
+#include <functional>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <locale>
+#include <sstream>
+#include <string>
+
+#include "ParamFile.hpp"
+
+// ltrim(), rtrim(), and trim() are from a highly up voted answer on StackOverflow
+// https://stackoverflow.com/a/217605
+
+// trim from start (in place)
+static inline void ltrim(std::string& s) {
```

Comment:
No need for `inline` here - its only a hint to the compiler - which will ignore it.